packages/web-react/src/components/FileUploader/FileUploaderAttachment.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { createRef, RefObject, useEffect } from 'react';
+import classNames from 'classnames';
+import { SpiritFileUploaderAttachmentProps } from '../../types';
+import { useStyleProps } from '../../hooks';
+import { useFileUploaderStyleProps } from './useFileUploaderStyleProps';
+import { Icon } from '../Icon';
+
+const FileUploaderAttachment = (props: SpiritFileUploaderAttachmentProps) => {
+  const { label, name, id, onDismiss, buttonLabel = 'Remove', file, ...restProps } = props;
+
+  const { classProps } = useFileUploaderStyleProps();
+  const { styleProps, props: transferProps } = useStyleProps(restProps);
+
+  const itemReference = createRef() as RefObject<HTMLLIElement>;
+
+  const createAttachmentInput = () => {
```

Comment:
This should be the React component. So you should add new components using a map based on queue content.

Reply:
On the one hand, you are right that in React scope it should be like a React component, but given that there is no way to "reach" into the element and set the file for it. CreatElement allows only element type, props and children. On the other hand, it's just a matter of taking the files from the inputs of these attachments when sending the form

Reply:
I do not understand. You cannot do `<input files={dataContainer.files} />`?

Reply:
No, it will appear as regular prop -> `files="[Object object]"`

Reply:
But it doesn't matter that it's not a React component, because you're just collecting the files from that queue anyway

Reply:
You have to understand what's going on there. Drag and drop is bound to the drop zone element, not to the input, so you handle onChange on the input and drag and drop events on the drop zone.

In addition, this input does not have a `name`, so it seems to be thrown out of form events. This name is just passed to attachments as `name[]`.

So I created a reference to both the input and the drop zone. So if needed, both references are available there. Examples are in both forms in the Storybook.

Rather than a reference to this input, I'm thinking of somehow outputting another field only with references for attachments, or extending the Map attachment object. But in my opinion, there is no need to solve it anyway, because you always have the fileQueue state where the files are

Reply:
Ok, this is refactored.